File: 2025/01/01.ts
import * as fs from "fs";

let input = fs.readFileSync("main.in").toString().split("\n");

part1(input);
part2(input);

function part1(input) {
  let pos = 50;
  let pwd = 0;
  for (const line of input) {
    let rot = Number(line.substring(1));
    if (line[0] === "L") {
      pos -= rot;
    } else {
      pos += rot;
    }
    if (pos % 100 === 0) {
      pwd += 1;
    }
  }
  console.log(pwd);
}

function part2(input) {
  let pos = 50;
  let pwd = 0;
  for (const line of input) {
    let rot = Number(line.substring(1));
    for (let i = 0; i < rot; i++) {
      if (line[0] === "L") {
        pos -= 1;
      } else {
        pos += 1;
      }
      if (pos % 100 === 0) {
        pwd += 1;
      }
    }
  }
  console.log(pwd);
}

File: 2020/01/01.py
numbers = []
while True:
    line = input()
    if not line:
        break
    numbers.append(int(line))

for i in range(len(numbers)):
    for j in range(i + 1, len(numbers)):
        if i != j and numbers[i] + numbers[j] == 2020:
            print(numbers[i] * numbers[j])

for i in range(len(numbers)):
    for j in range(i, len(numbers)):
        for k in range(j, len(numbers)):
            if numbers[i] + numbers[j] + numbers[k] == 2020:
                print(numbers[i] * numbers[j] * numbers[k])
